envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -182,6 +201,13 @@ data class TimeoutPolicy(
     val responseTimeout: Duration?
 )
 
+data class HealthCheck(
+    val path: String = "",
+    val clusterName: String = "local_service"
+) {
+    fun hasHealthCheck() = !path.isBlank()
```

Comment:
I don't get it, the default HC does not have a health check? (explaination: path="", path.isBlank() == true, hasHealthCheck = false).

Reply:
It is custom healthCheck. By default EC sets healthCheck to path defined in application properties and `local_service` cluster.

Reply:
I'll change the name to `hasCustomHealthCheck`.